api/db/migrations/20220721115757_alter-answers-id-from-int-to-bigint.js

```diff
@@ -0,0 +1,29 @@
+// In order to avoid alter bigint migration timeout, we check the number of answers in the target env before and run the alter bigint only if the limit is not exeeded.
+const maxAnswersToRunAlterMigration = 500000;
+
+const changeAnswerIdTypeToBigint = async (knex) => {
+  await knex.transaction(async (trx) => {
+    await knex.raw(`ALTER TABLE "knowledge-elements" ALTER COLUMN "answerId" TYPE BIGINT`).transacting(trx);
```

Comment:
Pourquoi on n'utilise pas les méthodes de knex ? 
```
return knex.schema.alterTable('knowledge-elements', function (table) {
    table.bigInt('answerId').alter();
  });
```

Reply:
J'ai testé avec la syntaxe knex proposé en rajoutant la transaction et bigInteger au lieu de bigInt.
J'ai eu des erreurs car visiblement cela enlève les contraintes sur la colonne et comme la colonne id de la table answer est une PK cela renvoie une erreur. Voir ci dessous les requêtes SQL générées par la syntaxe Knex, or ce qui est voulu est juste de changer le type.

knex:query alter table "knowledge-elements" alter column "answerId" drop default trx9 +2ms
  knex:bindings [] trx9 +2ms
  knex:query alter table "knowledge-elements" alter column "answerId" drop not null trx9 +2ms
  knex:bindings [] trx9 +2ms
  knex:query alter table "knowledge-elements" alter column "answerId" type bigint using ("answerId"::bigint) trx9 +1ms

Reply:
C'est étonnant, merci d'avoir essayé !